commit 90d5fa14dc67e3febe9ec3a714329ba911cc2acf
Author: Felipe Heidrich <fheidric>
Date:   Wed Feb 9 16:13:39 2011 +0000

    Bug 334147 - Several SWT snippets use incorrect positioning

3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet11.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet12.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet17.java
4	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet18.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet19.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet20.java
2	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet21.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet22.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet26.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet3.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet34.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet35.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet36.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet4.java
4	2	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet41.java
2	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet44.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet45.java
2	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet47.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet49.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet52.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet53.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet54.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet55.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet56.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet57.java
3	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet58.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet59.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet64.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet67.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet73.java
3	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet76.java
2	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet90.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet97.java
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet11.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet11.java
index e274805..dbfcbbd 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet11.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet11.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet11 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Text text = new Text (shell, SWT.BORDER | SWT.V_SCROLL);
-	text.setBounds (10, 10, 100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	text.setBounds (clientArea.x + 10, clientArea.y + 10, 100, 100);
 	for (int i=0; i<16; i++) {
 		text.append ("Line " + i + "\n");
 	}
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet12.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet12.java
index 8d7a992..f0de4af 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet12.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet12.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet12 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Text text = new Text (shell, SWT.BORDER | SWT.V_SCROLL);
-	text.setBounds (10, 10, 100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	text.setBounds (clientArea.x + 10, clientArea.y + 10, 100, 100);
 	for (int i=0; i<16; i++) {
 		text.append ("Line " + i + "\n");
 	}
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet17.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet17.java
index f0ff83b..b2cbc93 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet17.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet17.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet17 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Slider slider = new Slider (shell, SWT.HORIZONTAL);
-	slider.setBounds (10, 10, 200, 32);
+	Rectangle clientArea = shell.getClientArea ();
+	slider.setBounds (clientArea.x + 10, clientArea.y + 10, 200, 32);
 	slider.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event event) {
 			String string = "SWT.NONE";
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet18.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet18.java
index 808bfb9..bd1ab1b 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet18.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet18.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet18 {
@@ -28,10 +29,12 @@ public static void main (String [] args) {
 		ToolItem item = new ToolItem (bar, SWT.PUSH);
 		item.setText ("Item " + i);
 	}
+	Rectangle clientArea = shell.getClientArea ();
+	bar.setLocation (clientArea.x, clientArea.y);
 	bar.pack ();
 	shell.open ();
 	Display display = shell.getDisplay ();
-	while (!shell.isDisposed()) {
+	while (!shell.isDisposed ()) {
 		if (!display.readAndDispatch ()) display.sleep ();
 	}
 	display.dispose ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet19.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet19.java
index be69169..2f56a9c 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet19.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet19.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet19 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Text text = new Text (shell, SWT.BORDER | SWT.V_SCROLL);
-	text.setBounds (10, 10, 200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	text.setBounds ( clientArea.x + 10, clientArea.y + 10, 200, 200);
 	text.addListener (SWT.Verify, new Listener () {
 		public void handleEvent (Event e) {
 			String string = e.text;
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet20.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet20.java
index 46f0bd9..72f61d5 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet20.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet20.java
@@ -34,6 +34,8 @@ public static void main (String [] args) {
 		item.setPreferredSize (item.computeSize (size.x, size.y));
 		item.setControl (button);
 	}
+	Rectangle clientArea = shell.getClientArea ();
+	bar.setLocation (clientArea.x, clientArea.y);
 	bar.pack ();
 	shell.open ();
 	while (!shell.isDisposed ()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet21.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet21.java
index 317952d..406c4f6 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet21.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet21.java
@@ -28,7 +28,8 @@ public static void main (String [] args) {
 	final Color blue = display.getSystemColor (SWT.COLOR_BLUE);
 	Shell shell = new Shell (display);
 	Button b = new Button (shell, SWT.PUSH);
-	b.setBounds (10, 10, 100, 32);
+	Rectangle clientArea = shell.getClientArea ();
+	b.setBounds (clientArea.x + 10, clientArea.y + 10, 100, 32);
 	b.setText ("Button");
 	shell.setDefaultButton (b);
 	final Canvas c = new Canvas (shell, SWT.BORDER);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet22.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet22.java
index 765b5ba..c5f73e2 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet22.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet22.java
@@ -16,6 +16,7 @@ package org.eclipse.swt.snippets;
  * For a list of all SWT example snippets see
  * http://www.eclipse.org/swt/snippets/
  */
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet22 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Shell shell = new Shell (display);
 	Text text = new Text (shell, 0);
 	text.setText ("ASDF");
-	text.setSize (64, 32);
+	Rectangle clientArea = shell.getClientArea ();
+	text.setBounds (clientArea.x, clientArea.y, 64, 32);
 	text.selectAll ();
 	shell.pack ();
 	shell.open ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet26.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet26.java
index e1fbde1..dabc0be 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet26.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet26.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet26 {
@@ -26,7 +27,8 @@ public static void main (String [] args) {
 	Shell shell = new Shell (display);
 	Combo combo = new Combo (shell, SWT.READ_ONLY);
 	combo.setItems (new String [] {"Alpha", "Bravo", "Charlie"});
-	combo.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	combo.setBounds (clientArea.x, clientArea.y, 200, 200);
 	shell.pack ();
 	shell.open ();
 	while (!shell.isDisposed ()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet3.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet3.java
index a95f1ed..c3ab2c4 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet3.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet3.java
@@ -43,6 +43,8 @@ public static void main(String[] args) {
 	for (int i = 0; i < columnCount; i++) {
 		table.getColumn(i).pack();
 	}
+	Rectangle clientArea = shell.getClientArea();
+	table.setLocation(clientArea.x, clientArea.y);
 	Point size = table.computeSize(SWT.DEFAULT, 200);
 	table.setSize(size);
 	shell.pack();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet34.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet34.java
index 27765cb..87a6477 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet34.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet34.java
@@ -32,6 +32,8 @@ public static void main (String[] args) {
 	gc.dispose ();
 	Shell shell = new Shell (display);
 	Label label = new Label (shell, SWT.BORDER);
+	Rectangle clientArea = shell.getClientArea ();
+	label.setLocation (clientArea.x, clientArea.y);
 	label.setImage (image);
 	label.pack ();
 	shell.pack ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet35.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet35.java
index 38387ce..f79e184 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet35.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet35.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet35 {
@@ -29,7 +30,8 @@ public static void main (String [] args) {
 		TableItem item = new TableItem (table, 0);
 		item.setText ("Item " + i);
 	}
-	table.setSize (100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x, clientArea.y, 100, 100);
 	shell.setSize (200, 200);
 	shell.open ();
 	while (!shell.isDisposed()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet36.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet36.java
index 7d4919d..a7523d7 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet36.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet36.java
@@ -32,6 +32,8 @@ public static void main (String [] args) {
 	gc.dispose ();
 	Shell shell = new Shell (display);
 	ToolBar toolBar = new ToolBar (shell, SWT.FLAT | SWT.BORDER);
+	Rectangle clientArea = shell.getClientArea ();
+	toolBar.setLocation (clientArea.x, clientArea.y);
 	for (int i=0; i<12; i++) {
 		int style = i % 3 == 2 ? SWT.DROP_DOWN : SWT.PUSH;
 		ToolItem item = new ToolItem (toolBar, style);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet4.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet4.java
index 8fc8411..453b851 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet4.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet4.java
@@ -18,6 +18,7 @@ package org.eclipse.swt.snippets;
  */
 import org.eclipse.swt.*;
 import org.eclipse.swt.events.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet4 {
@@ -28,7 +29,8 @@ public class Snippet4 {
 		Button b = new Button(shell, SWT.PUSH);
 		b.setText("Open Dialog ...");
 		b.pack();
-		b.setLocation(10, 10);
+		Rectangle clientArea = shell.getClientArea();
+		b.setLocation(clientArea.x + 10, clientArea.y + 10);
 		b.addSelectionListener(new SelectionAdapter() {
 			public void widgetSelected(SelectionEvent se) {
 				Shell dialog = new Shell(shell, SWT.DIALOG_TRIM);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet41.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet41.java
index 3703753..f63ea90 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet41.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet41.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet41 {
@@ -26,12 +27,13 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	TabFolder folder = new TabFolder (shell, SWT.BORDER);
-	folder.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	folder.setBounds (clientArea.x, clientArea.y, 200, 200);
 	TabItem item0 = new TabItem (folder, 0);
 	item0.setText ("tab item");
 	item0.setToolTipText ("TabItem toolTip: " + string);
 	ToolBar bar = new ToolBar (shell, SWT.BORDER);
-	bar.setBounds (0, 200, 200, 40);
+	bar.setBounds (clientArea.x, clientArea.y + 200, 200, 40);
 	ToolItem item1 = new ToolItem (bar, SWT.PUSH);
 	item1.setText ("tool item");
 	item1.setToolTipText ("ToolItem toolTip: " + string);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet44.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet44.java
index 67ee317..c405a49 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet44.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet44.java
@@ -29,7 +29,8 @@ public static void main (String [] args) {
 	shell.open ();
 	final Button b = new Button (shell, 0);
 	b.setText("Push to set cursor to hand");
-	b.setBounds (10, 10, 200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	b.setBounds (clientArea.x + 10, clientArea.y + 10, 200, 200);
 	b.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event e) {
 			b.setCursor (cursor);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet45.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet45.java
index 52acee3..eb14486 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet45.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet45.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet45 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Scale scale = new Scale (shell, SWT.BORDER);
-	scale.setSize (200, 64);
+	Rectangle clientArea = shell.getClientArea ();
+	scale.setBounds (clientArea.x, clientArea.y, 200, 64);
 	scale.setMaximum (40);
 	scale.setPageIncrement (5);
 	shell.open ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet47.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet47.java
index 16055ca..28e2237 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet47.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet47.java
@@ -48,7 +48,8 @@ public static void main (String [] args) {
 	gc.dispose ();
 	
 	ToolBar bar = new ToolBar (shell, SWT.BORDER | SWT.FLAT);
-	bar.setSize (200, 32);
+	Rectangle clientArea = shell.getClientArea ();
+	bar.setBounds (clientArea.x, clientArea.y, 200, 32);
 	for (int i=0; i<12; i++) {
 		ToolItem item = new ToolItem (bar, 0);
 		item.setImage (image);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet49.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet49.java
index b66a128..ddd436d 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet49.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet49.java
@@ -37,6 +37,8 @@ public static void main (String [] args) {
 			toolBar.setSize (size);
 		}
 	});
+	Rectangle clientArea = shell.getClientArea ();
+	toolBar.setLocation (clientArea.x, clientArea.y);
 	toolBar.pack ();
 	shell.pack ();
 	shell.open ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet52.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet52.java
index 3442460..9a2a1b4 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet52.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet52.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet52 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Table table = new Table (shell, SWT.BORDER | SWT.MULTI);
-	table.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x, clientArea.y, 200, 200);
 	for (int i=0; i<128; i++) {
 		TableItem item = new TableItem (table, SWT.NONE);
 		item.setText ("Item " + i);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet53.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet53.java
index bca0148..1a9e35b 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet53.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet53.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet53 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	final Table table = new Table (shell, SWT.BORDER | SWT.MULTI);
-	table.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x, clientArea.y, 200, 200);
 	for (int i=0; i<128; i++) {
 		TableItem item = new TableItem (table, SWT.NONE);
 		item.setText ("Item " + i);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet54.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet54.java
index 0622afa..bfcc4d6 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet54.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet54.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet54 {
@@ -26,7 +27,8 @@ public static void main (String [] args) {
 	Shell shell = new Shell (display);
 	shell.setSize(400, 300);
 	final Sash sash = new Sash (shell, SWT.BORDER | SWT.VERTICAL);
-	sash.setBounds (180, 10, 32, 240);
+	Rectangle clientArea = shell.getClientArea ();
+	sash.setBounds (180, clientArea.y, 32, clientArea.height);
 	sash.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event e) {
 			sash.setBounds (e.x, e.y, e.width, e.height);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet55.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet55.java
index 0e6a83f..b53a814 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet55.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet55.java
@@ -26,6 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Text text = new Text (shell, SWT.BORDER);
+	Rectangle clientArea = shell.getClientArea ();
+	text.setLocation (clientArea.x, clientArea.y);
 	int columns = 10;
 	GC gc = new GC (text);
 	FontMetrics fm = gc.getFontMetrics ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet56.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet56.java
index 9eeabb8..ba5137a 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet56.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet56.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet56 {
@@ -25,7 +26,8 @@ public class Snippet56 {
 		final Display display = new Display();
 		Shell shell = new Shell(display);
 		final ProgressBar bar = new ProgressBar(shell, SWT.SMOOTH);
-		bar.setBounds(10, 10, 200, 32);
+		Rectangle clientArea = shell.getClientArea ();
+		bar.setBounds (clientArea.x, clientArea.y, 200, 32);
 		shell.open();
 		final int maximum = bar.getMaximum();
 		new Thread() {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet57.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet57.java
index 10b9d8f..a801c27 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet57.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet57.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet57 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	final Display display = new Display ();
 	Shell shell = new Shell (display);
 	final ProgressBar bar = new ProgressBar (shell, SWT.SMOOTH);
-	bar.setBounds (10, 10, 200, 32);
+	Rectangle clientArea = shell.getClientArea ();
+	bar.setBounds (clientArea.x, clientArea.y, 200, 32);
 	shell.open ();
 	
 	display.timerExec(100, new Runnable() {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet58.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet58.java
index a914414..ec7ee66 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet58.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet58.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet58 {
@@ -25,6 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	ToolBar bar = new ToolBar (shell, SWT.BORDER);
+	Rectangle clientArea = shell.getClientArea ();
+	bar.setLocation (clientArea.x, clientArea.y);
 	for (int i=0; i<4; i++) {
 		ToolItem item = new ToolItem (bar, 0);
 		item.setText ("Item " + i);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet59.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet59.java
index 978dacd..45df7c4 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet59.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet59.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet59 {
@@ -26,7 +27,8 @@ public static void main (String [] args) {
 	Shell shell = new Shell (display);
 	final List list = new List (shell, SWT.BORDER | SWT.MULTI | SWT.V_SCROLL);
 	for (int i=0; i<128; i++) list.add ("Item " + i);
-	list.setBounds (0, 0, 100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	list.setBounds (clientArea.x, clientArea.y, 100, 100);
 	list.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event e) {
 			String string = "";
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet64.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet64.java
index 8a8db13..06b6797 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet64.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet64.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet64 {
@@ -29,7 +30,8 @@ public static void main (String [] args) {
 		TableItem item = new TableItem (table, 0);
 		item.setText ("Item " + i);
 	}
-	table.setBounds (0, 0, 100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x, clientArea.y, 100, 100);
 	table.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event e) {
 			String string = "";
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet67.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet67.java
index 0214e9d..a809f6e 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet67.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet67.java
@@ -26,6 +26,8 @@ public static void main (String [] args) {
 	final Display display = new Display ();
 	final Shell shell = new Shell (display);
 	final ToolBar toolBar = new ToolBar (shell, SWT.NONE);
+	Rectangle clientArea = shell.getClientArea ();
+	toolBar.setLocation(clientArea.x, clientArea.y);
 	final Menu menu = new Menu (shell, SWT.POP_UP);
 	for (int i=0; i<8; i++) {
 		MenuItem item = new MenuItem (menu, SWT.PUSH);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet73.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet73.java
index 45d855c..e74646b 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet73.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet73.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet73 {
@@ -48,7 +49,8 @@ public static void main (String [] args) {
 			}
 		}
 	});
-	tree.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	tree.setBounds (clientArea.x, clientArea.y, 200, 200);
 	shell.setSize (300, 300);
 	shell.open ();
 	while (!shell.isDisposed ()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet76.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet76.java
index e81783b..aa7b1a2 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet76.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet76.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet76 {
@@ -25,6 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	final Shell shell = new Shell (display);
 	final TabFolder tabFolder = new TabFolder (shell, SWT.BORDER);
+	Rectangle clientArea = shell.getClientArea ();
+	tabFolder.setLocation (clientArea.x, clientArea.y);
 	for (int i=0; i<6; i++) {
 		TabItem item = new TabItem (tabFolder, SWT.NONE);
 		item.setText ("TabItem " + i);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet90.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet90.java
index 7bf13cd..a39a5b1 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet90.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet90.java
@@ -39,7 +39,8 @@ public static void main (String [] args) {
 			}
 		}
 	});
-	tree.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	tree.setBounds (clientArea.x, clientArea.y, 200, 200);
 	shell.setSize (300, 300);
 	shell.open ();
 	while (!shell.isDisposed ()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet97.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet97.java
index 1d0174c..21e7b21 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet97.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet97.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet97 {
@@ -44,7 +45,8 @@ public static void main (String [] args) {
 			}
 		}
 	});
-	tree.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	tree.setBounds (clientArea.x, clientArea.y, 200, 200);
 	shell.setSize (300, 300);
 	shell.open ();
 	while (!shell.isDisposed ()) {
commit 101250c45d80fc34275150d25ceaef92cfc2d345
Author: Felipe Heidrich <fheidric>
Date:   Mon Feb 14 15:39:17 2011 +0000

    Bug 334147 - Several SWT snippets use incorrect positioning

2	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet100.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet101.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet102.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet103.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet110.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet113.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet114.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet182.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet183.java
3	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet184.java
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet100.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet100.java
index 2fff80a..63ce3d6 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet100.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet100.java
@@ -27,7 +27,8 @@ public static void main(String[] args) {
 	Shell shell = new Shell(display);
 	shell.setBounds(10, 10, 200, 200);
 	Text text = new Text(shell, SWT.MULTI);
-	text.setBounds(10, 10, 150, 150);
+	Rectangle clientArea = shell.getClientArea();
+	text.setBounds(clientArea.x + 10, clientArea.y + 10, 150, 150);
 	Font initialFont = text.getFont();
 	FontData[] fontData = initialFont.getFontData();
 	for (int i = 0; i < fontData.length; i++) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet101.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet101.java
index d70e976..1292066 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet101.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet101.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet101 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Table table = new Table (shell, SWT.BORDER | SWT.MULTI);
-	table.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x, clientArea.y, 200, 200);
 	for (int i=0; i<12; i++) {
 		TableItem item = new TableItem (table, SWT.NONE);
 		item.setText ("Item " + i);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet102.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet102.java
index 23056b6..38df07e 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet102.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet102.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet102 {
@@ -25,7 +26,8 @@ public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Tree tree = new Tree (shell, SWT.BORDER | SWT.MULTI);
-	tree.setSize (200, 200);
+	Rectangle clientArea = shell.getClientArea ();
+	tree.setBounds (clientArea.x, clientArea.y, 200, 200);
 	for (int i=0; i<12; i++) {
 		TreeItem item = new TreeItem (tree, SWT.NONE);
 		item.setText ("Item " + i);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet103.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet103.java
index 5ab73fb..c8fa8c3 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet103.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet103.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet103 {
@@ -27,7 +28,8 @@ public static void main(String[] args) {
 	Shell shell = new Shell(display);
 	shell.setBounds(10, 10, 200, 240);
 	Table table = new Table(shell, SWT.NONE);
-	table.setBounds(10, 10, 160, 160);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x + 10, clientArea.y + 10, 160, 160);
 
 	final TableItem[] items = new TableItem[4];
 	for (int i = 0; i < 4; i++) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet110.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet110.java
index 5bda0c3..084d032 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet110.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet110.java
@@ -42,6 +42,8 @@ public static void main (String [] args) {
 	for (int i=0; i < columnCount; i++) {
 		table.getColumn (i).pack ();
 	}
+	Rectangle clientArea = shell.getClientArea ();
+	table.setLocation (clientArea.x, clientArea.y);
 	Point size = table.computeSize (SWT.DEFAULT, 200);
 	table.setSize (size);
 	shell.pack ();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet113.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet113.java
index 87ba34e..bf43233 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet113.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet113.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet113 {
@@ -29,7 +30,8 @@ public static void main (String [] args) {
 		TableItem item = new TableItem (table, SWT.NONE);
 		item.setText ("Item " + i);
 	}
-	table.setSize (100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	table.setBounds (clientArea.x, clientArea.y, 100, 100);
 	table.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event event) {
 			String string = event.detail == SWT.CHECK ? "Checked" : "Selected";
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet114.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet114.java
index cec6fc6..b0c1f79 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet114.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet114.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet114 {
@@ -29,7 +30,8 @@ public static void main (String [] args) {
 		TreeItem item = new TreeItem (tree, SWT.NONE);
 		item.setText ("Item " + i);
 	}
-	tree.setSize (100, 100);
+	Rectangle clientArea = shell.getClientArea ();
+	tree.setBounds (clientArea.x, clientArea.y, 100, 100);
 	tree.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event event) {
 			String string = event.detail == SWT.CHECK ? "Checked" : "Selected";
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet182.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet182.java
index f710f21..32fe5f3 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet182.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet182.java
@@ -19,6 +19,7 @@ package org.eclipse.swt.snippets;
  * @since 3.1
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet182 {
@@ -28,7 +29,8 @@ public class Snippet182 {
 		Shell shell = new Shell(display);
 		Link link = new Link(shell, SWT.BORDER);
 		link.setText("This a very simple <A>link</A> widget.");
-		link.setSize(140, 40);
+		Rectangle clientArea = shell.getClientArea();
+		link.setBounds(clientArea.x, clientArea.y, 140, 40);
 		shell.pack ();
 		shell.open();
 		while (!shell.isDisposed()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet183.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet183.java
index 88c5a1e..0301550 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet183.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet183.java
@@ -19,6 +19,7 @@ package org.eclipse.swt.snippets;
  * @since 3.1
  */
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet183 {
@@ -29,7 +30,8 @@ public class Snippet183 {
 		Link link = new Link(shell, SWT.NONE);
 		String text = "The SWT component is designed to provide <a>efficient</a>, <a>portable</a> <a href=\"native\">access to the user-interface facilities of the operating systems</a> on which it is implemented.";
 		link.setText(text);
-		link.setSize(400, 400);
+		Rectangle clientArea = shell.getClientArea();
+		link.setBounds(clientArea.x, clientArea.y, 400, 400);
 		link.addListener (SWT.Selection, new Listener () {
 			public void handleEvent(Event event) {
 				System.out.println("Selection: " + event.text);
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet184.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet184.java
index 331895a..9f5ddfc1 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet184.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet184.java
@@ -19,6 +19,7 @@ package org.eclipse.swt.snippets;
  * @since 3.1
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet184 {
@@ -32,6 +33,8 @@ public class Snippet184 {
 		spinner.setSelection(500);
 		spinner.setIncrement(1);
 		spinner.setPageIncrement(100);
+		Rectangle clientArea = shell.getClientArea();
+		spinner.setLocation(clientArea.x, clientArea.y);
 		spinner.pack();
 		shell.pack();
 		shell.open();
commit bdf4f9615d196b856239c88f144a55022b23fb8b
Author: Felipe Heidrich <fheidric>
Date:   Thu Feb 17 16:51:26 2011 +0000

    Bug 334147 - Several SWT snippets use incorrect positioning

2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet225.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet238.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet241.java
2	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet245.java
5	3	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet249.java
3	1	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet274.java
2	0	examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet276.java
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet225.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet225.java
index e77752c..7289c02 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet225.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet225.java
@@ -48,6 +48,8 @@ public static void main(String[] args) {
 			tip.setVisible(true);
 		}
 	});
+	Rectangle clientArea = shell.getClientArea();
+	button.setLocation(clientArea.x, clientArea.y);
 	button.pack();
 	shell.setBounds(50, 50, 300, 200);
 	shell.open();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet238.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet238.java
index ca6ce04..c330fa3 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet238.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet238.java
@@ -17,6 +17,7 @@ package org.eclipse.swt.snippets;
  * http://www.eclipse.org/swt/snippets/
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet238 {
@@ -25,7 +26,8 @@ public static void main(String[] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
 	Composite composite = new Composite (shell, SWT.BORDER);
-	composite.setSize (100, 100);
+	Rectangle clientArea = shell.getClientArea();
+	composite.setBounds (clientArea.x, clientArea.y, 100, 100);
 	Menu menu = new Menu (shell, SWT.POP_UP);
 	MenuItem item1 = new MenuItem (menu, SWT.PUSH);
 	item1.setText ("Push Item");
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet241.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet241.java
index 1026273..2d34c0a 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet241.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet241.java
@@ -19,6 +19,7 @@ package org.eclipse.swt.snippets;
 
 import org.eclipse.swt.*;
 import org.eclipse.swt.events.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet241 {
@@ -28,7 +29,8 @@ public static void main(String [] args) {
 	Shell shell = new Shell(display);
 	shell.setBounds(10,10,200,200);
 	Text text1 = new Text(shell, SWT.MULTI | SWT.WRAP);
-	text1.setBounds(10,10,150,50);
+	Rectangle clientArea = shell.getClientArea();
+	text1.setBounds(clientArea.x+10,clientArea.y+10,150,50);
 	text1.setText("Tab will traverse out from here.");
 	text1.addTraverseListener(new TraverseListener() {
 		public void keyTraversed(TraverseEvent e) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet245.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet245.java
index 00b8425..67c2739 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet245.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet245.java
@@ -32,7 +32,8 @@ public static void main(String [] args) {
 			event.gc.drawOval(0, 0, rect.width - 1, rect.height - 1);
 		}
 	});
-	shell.setBounds(10, 10, 200, 200);
+	Rectangle clientArea = shell.getClientArea();
+	shell.setBounds(clientArea.x + 10, clientArea.y + 10, 200, 200);
 	shell.open ();
 	while (!shell.isDisposed()) {
 		if (!display.readAndDispatch()) display.sleep();
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet249.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet249.java
index 6f24ab9..204820d 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet249.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet249.java
@@ -18,6 +18,7 @@ package org.eclipse.swt.snippets;
  */
 import org.eclipse.swt.*;
 import org.eclipse.swt.custom.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.layout.*;
 import org.eclipse.swt.widgets.*;
 
@@ -28,10 +29,11 @@ public class Snippet249 {
 public static void main (String [] args) {
 	Display display = new Display ();
 	Shell shell = new Shell (display);
-	shell.setBounds (10, 10, 300, 200);
+	Rectangle clientArea = shell.getClientArea();
+	shell.setBounds (clientArea.x + 10, clientArea.y + 10, 300, 200);
 	// create the composite that the pages will share
 	final Composite contentPanel = new Composite (shell, SWT.BORDER);
-	contentPanel.setBounds (100, 10, 190, 90);
+	contentPanel.setBounds (clientArea.x + 100, clientArea.y + 10, 190, 90);
 	final StackLayout layout = new StackLayout ();
 	contentPanel.setLayout (layout);
 
@@ -52,7 +54,7 @@ public static void main (String [] args) {
 	// create the button that will switch between the pages
 	Button pageButton = new Button (shell, SWT.PUSH);
 	pageButton.setText ("Push");
-	pageButton.setBounds (10, 10, 80, 25);
+	pageButton.setBounds (clientArea.x + 10, clientArea.y + 10, 80, 25);
 	pageButton.addListener (SWT.Selection, new Listener () {
 		public void handleEvent (Event event) {
 			pageNum = ++pageNum % 2;
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet274.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet274.java
index f84234d..7246a53 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet274.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet274.java
@@ -19,6 +19,7 @@ package org.eclipse.swt.snippets;
  * @since 3.3
  */
 import org.eclipse.swt.*;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.widgets.*;
 
 public class Snippet274 {
@@ -79,7 +80,8 @@ public class Snippet274 {
 	            }
 	        }
 	    }
-	    tree.setSize(200, 200);
+		Rectangle clientArea = shell.getClientArea();
+	    tree.setBounds(clientArea.x, clientArea.y, 200, 200);
 	    shell.pack();
 	    shell.open();
 	    while (!shell.isDisposed()) {
diff --git a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet276.java b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet276.java
index ab779f6..2ae5afb 100644
--- a/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet276.java
+++ b/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet276.java
@@ -36,6 +36,8 @@ public static void main (String[] args) {
 	};
 	shell.addListener (SWT.MouseDown, listener);
 	label.addListener (SWT.MouseDown, listener);
+	Rectangle clientArea = shell.getClientArea();
+	label.setLocation(clientArea.x, clientArea.y);
 	label.pack ();
 	shell.open ();
 	while (!shell.isDisposed ()){
